python/packages/durabletask/DESIGN.md

```diff
@@ -0,0 +1,249 @@
+# Design: Durable Task Provider for Agent Framework
+
+## Overview
+
+This package, `agent-framework-durabletask`, provides a durability layer for the Microsoft Agent Framework using the `durabletask` Python SDK. It enables stateful, reliable, and distributed agent execution on any platform (Bring Your Own Platform), decoupling the agent's durability from the Azure Functions platform.
+
+## Design Decision
+
+**Selected Approach: Object-Oriented Wrappers with Symmetric Factory Pattern**
+
+We will use a symmetric Object-Oriented design where both the Client (external) and Orchestrator (internal) expose a consistent interface for retrieving and interacting with durable agents.
+
+## Core Philosophy
+
+*   **Native `DurableEntity` Support**: We will leverage the `DurableEntity` support introduced in `durabletask` v1.0.0.
+*   **Symmetric Factories**: `DurableAIAgentClient` (for external use) and `DurableAIAgentOrchestrator` (for internal use) both provide a `get_agent` method.
+*   **Unified Interface**: `DurableAIAgent` serves as the common interface for executing agents, regardless of the context (Client vs Orchestration).
+*   **Consistent Return Type**: `DurableAIAgent.run` always returns a `Task` (or compatible awaitable), ensuring consistent usage patterns.
+
+## Architecture
+
+### 1. Package Structure
+
+```text
+packages/durabletask/
+├── pyproject.toml
+├── README.md
+├── agent_framework_durabletask/
+│   ├── __init__.py
+│   ├── _worker.py      # DurableAIAgentWorker
+│   ├── _client.py      # DurableAIAgentClient
+│   ├── _orchestrator.py # DurableAIAgentOrchestrator
+│   ├── _entities.py    # AgentEntity implementation
+│   ├── _models.py      # Data models (RunRequest, AgentResponse, etc.)
+│   ├── _durable_agent_state.py # State schema (Ported from azurefunctions)
+│   ├── _shim.py        # DurableAIAgent implementation (will be ported from azurefunctions)
+│   └── _utils.py       # Mixins and helpers
+└── tests/
+```
+
+### 2. State Management (`_durable_agent_state.py`)
+
+*   **Goal**: Maintain 100% schema compatibility with `agent-framework-azurefunctions`.
+*   **Implementation**: Direct port of `packages/azurefunctions/agent_framework_azurefunctions/_durable_agent_state.py`.
```

Comment:
The phrasing of these two points is weird because it implies that these are two different products that need to be kept in sync. In reality, one is just a wrapper SDK for the other.

Reply:
Reworded